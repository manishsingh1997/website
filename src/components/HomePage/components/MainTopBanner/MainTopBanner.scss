@import '@ergeon/core-components/src/components/variables.scss';

$banner-desktop-height: $h-60 * 10;
$banner-mobile-height: $banner-desktop-height / 2;
$banner-top-desktop-padding: $v-88 + $v-12;

@mixin use-title-font {
  font-family: $font-title;
  font-size: $h-20 + 2;
}

.MainTopBanner {
  background-image: url('../../../../assets/fence-background.jpg');
  background-size: cover;
  background-position: center;
  display: flex;
  flex-direction: column;
  height: $banner-desktop-height;
  padding-top: $banner-top-desktop-padding;
  width: 100%;

  @include phone() {
    background-position: unset;
    height: $banner-mobile-height;
    padding-top: $v-48;
  }

  &-Content {
    color: $white;

    @include phone() {
      padding: 0 $h-20;
      text-align: center;
    }

    .CitySearchInput {
      @include phone() {
        margin: 0 auto;
      }
    }

    &--Heading {
      font-weight: 600;
      color: $white;
      margin-bottom: $v-24;

      @include phone() {
        font-size: $h4-size;
        line-height: $h4-line-height;
        margin: $v-16 auto;
        width: 70%;
      }
    }

    &--Subheading {
      margin-bottom: $v-12;
    }
  }
}
