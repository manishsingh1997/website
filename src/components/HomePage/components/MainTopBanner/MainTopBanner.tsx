
import React from 'react';

import CitySearch from '../CitySearch';

import './MainTopBanner.scss';

const MainTopBanner = () => {

  return (
    <section className="MainTopBanner">
      <div className="MainTopBanner-Content">
        <div className="wrapper-1180">
          <h2 className="MainTopBanner-Content--Heading">
            Improving your outdoor<br /> space can be easy & friendly
          </h2>
          <p className="MainTopBanner-Content--SubHeading">
            Find Ergeon fence expertise in your area
          </p>
          <CitySearch />
        </div>
      </div>
    </section>
  )
}

export default MainTopBanner;
