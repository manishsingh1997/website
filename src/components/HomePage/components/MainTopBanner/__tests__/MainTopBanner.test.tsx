import React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import MainTopBanner from '../MainTopBanner';

jest.mock('../../CitySearch', () => () => <>City search input</>);

describe('MainTopBanner', () => {
  it('should match snapshot', async () => {
    const { container } = render(<MainTopBanner />);
    expect(container).toMatchSnapshot();
  });
});
