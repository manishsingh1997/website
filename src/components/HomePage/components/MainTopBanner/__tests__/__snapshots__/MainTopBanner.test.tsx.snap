// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTopBanner should match snapshot 1`] = `
<div>
  <section
    class="MainTopBanner"
  >
    <div
      class="MainTopBanner-Content"
    >
      <div
        class="wrapper-1180"
      >
        <h1
          class="h2 MainTopBanner-Content--Heading"
        >
          Improving your outdoor
          <br />
           space can be easy & friendly
        </h1>
        <p
          class="MainTopBanner-Content--SubHeading"
        >
          Find Ergeon fence expertise in your area
        </p>
        City search input
      </div>
    </div>
  </section>
</div>
`;
